### PR TITLE
allow task based async api

### PR DESCRIPTION
**Pull request by coolya**
Exposed a task based API in the new  module `Task`. It uses  `TaskBuilder.fs` for the computational expressions. The original API still works as before but calls into the task base API and then converts it into async workflows. 

Especially in use cases where the calling code already uses tasks like for instance when using Giraffe it feels much more natural to have a task based API rather then converting the async workflows into tasks again. Since the API of the TableClient already exposes tasks the conversion from task to async only happens when caller is using Async or when using synchronous calls.

### CODE REVIEW

**Comment by daniel-chambers** [comment]
Thanks for the PR! I'm away this weekend, but I'll take a look next week
when I'm home. :)

**Comment by coolya** [comment]
Thanks for the quick notice :) No need to hurry and have a nice weekend. 👍

File: src/FSharp.Azure.Storage/Table.fs [hunk 1]
```diff
@@ -363,25 +363,65 @@ module Table =
...
-            return result |> convertToOperationResult
-        }
+            let tableQuery = query.ToTableQuery()
+            let resolver = EntityTypeCache.Resolver.Value //Do not inline this otherwise FSharp will delay execution of .Value until the resolver delegate is called
+            task {
+                let! result = table.ExecuteQuerySegmentedAsync<'T * EntityMetadata>(tableQuery, EntityResolver(resolver), continuationToken |> toNullRef)
+                return result.Results, result.ContinuationToken |> toOption
+            }
+        let fromTableAsync (client: CloudTableClient) tableName (query : EntityQuery<'T>) =
+            let rec getSegmentAsync continutationToken resultsLength resultsList =
```
[the remark this answers is not in the document]
Comment (coolya) [reply src/FSharp.Azure.Storage/Table.fs]: Good spot, totally overlooked that one. Addressing it now.

File: paket.dependencies [hunk 2]
```diff
@@ -3,6 +3,7 @@ framework: auto-detect
 
 //Main deps
 nuget FSharp.Core >= 4.1
+nuget TaskBuilder.fs 2.1.0
```
[the remark this answers is not in the document]
Comment (coolya) [reply paket.dependencies]: I think using `>= 2.1.0` should be fine.

**Comment by coolya** [comment]
I have reworked the `fromTableAsync` implementation to no longer use recursion. I don't really like the implementation because it is not really fsharp idiomatic and uses a mutable list to work around limitations of task builder but I guess it's the only way to get it working.

**Pull request merged**